resource-definitions/terraform-driver/co-provision/s3-co-provision.yaml

```diff
@@ -0,0 +1,49 @@
+apiVersion: entity.humanitec.io/v1b1
+kind: Definition
+metadata:
+  id: s3-co-provision
+entity:
+  name: s3-co-provision
+  type: s3
+  driver_type: humanitec/terraform
+  # Use the credentials injected via the driver_account to set variables as expected by your Terraform code
+  driver_account: aws
+  driver_inputs:
+    values:
+      variables:
+        REGION: eu-central-1
+      credentials_config:
+        variables:
+          ACCESS_KEY_ID: AccessKeyId
+          ACCESS_KEY_VALUE: SecretAccessKey
+      script: |-
+        # This provider block is using the Terraform variables
+        # set through the credentials_config.
+        # Variable declarations omitted for brevity.
+        provider "aws" {
+          region     = var.REGION
+          access_key = var.ACCESS_KEY_ID
+          secret_key = var.ACCESS_KEY_VALUE
+        }
+
+        # ... Terraform code reduced for brevity
+
+        resource "aws_s3_bucket" "bucket" {
+          bucket = my-bucket
+        }
+        
+        output "bucket" {
+          value = aws_s3_bucket.main.id
+        }
+  
+        output "arn" {
+          value = aws_s3_bucket.main.arn
+        }
+          
+        output "region" {
+          value = aws_s3_bucket.main.region
+        }
+  # Co-provision aws-policy resource
+  provision:
+    aws-policy:
+      is_dependant: false
```

Comment:
This should be `is_dependent`.
We don't set this flag to `true` as the dependency is explicit in the policy definition? I'm asking as, actually, `is_dependent` is true here.

Reply:
I think in this example it doesn't really matter, because there's a placeholder dependency anyway. In our demos and previous examples, we used is_dependent:false in this particular case, that's why I took it from there. But happy to change it, if you think it's more explanatory to use `true`.

Reply:
I think it depends on how we think about `co_provision.is_dependent` meaning. 
Anyweay, if we want to use it when the dependency is not explicitly defined by a placeholder usage, maybe we should just leave the field empty, so:
```yaml
  provision:
    aws-policy:
```
Without highlighting a non-dependency that is then created in the aws policy resource definition via placeholder.

Reply:
Yes, but maybe as it is an example and we want to show it's features, it's better to set `is_dependent"true` explicitly.

Reply:
Updated to "true"

Reply:
> I think it depends on how we think about co_provision.is_dependent meaning.
Good point. What is the meaning, formally? I checked and we do not define it in our [documentation on co-provisioning](https://developer.humanitec.com/platform-orchestrator/resources/resource-graph/#co-provision-resources), which we should.

My understanding is that `is_dependent: false` means that the co-provisioning resource (here: the `s3`) does not depend on the co-provisioned resource (the `aws-policy`). It has to be `false` or else we would have a circular dependency.

Is that correct?

Reply:
👍 Just one thing: it should be `is_dependent` while here it is  still `is_dependant`.

Reply:
<img width="687" alt="image" src="https://github.com/humanitec-architecture/example-library/assets/140624220/08d0fe3d-bf96-406d-99cf-4868b84068bf">

To illustrate, we can add this little mermaid diagram. I am pushing it to the README...

Reply:
It's not so obvious what is `R`, do we need it in the diagram, or maybe change it to Workload 🤔

Reply:
> It's not so obvious what is R

I saw you added an explanation, sound good!

Reply:
Yes sorry for the multiple commits, the chart preview I get locally is not identical to what GitHub makes of it, so I have to iterate somewhat. "Workload" is a lot better than "R" - changed that, and done now!